src/mono/mono/metadata/cominterop.c

```diff
@@ -2104,7 +2104,7 @@ cominterop_get_ccw_method (MonoClass *iface, MonoMethod *method, MonoError *erro
 	cominterop_setup_marshal_context (&m, adjust_method);
 	m.mb = mb;
 	m.runtime_marshalling_enabled = TRUE;
-	mono_marshal_emit_managed_wrapper (mb, sig_adjusted, mspecs, &m, adjust_method, 0);
+	mono_marshal_emit_managed_wrapper (mb, sig_adjusted, mspecs, &m, adjust_method, 0, NULL);
```

Comment:
@lateralusX are we going to build on Windows-like platforms with COM support enabled? You may want to also pick up https://github.com/mono/mono/pull/21365 which fixes a coop handle leak which leads to incorrect refcounts on some CCW objects

Reply:
For now I will keep it disabled, but maybe we should include that PR anyways? Do you want me to include it in this PR?

Reply:
I fixed the use of mono_marshal_emit_managed_wrapper to always require a MonoError.

Reply:
> For now I will keep it disabled, but maybe we should include that PR anyways? Do you want me to include it in this PR?

Yea, I think let's pick up those changes.

Long-term I'm not sure if we should re-enable cominterop.c in dotnet/runtime.  I think we're better off implementing ComWrappers.  But I suppose that depends on the use-cases.

Reply:
> Long-term I'm not sure if we should re-enable cominterop.c in dotnet/runtime. I think we're better off implementing ComWrappers. But I suppose that depends on the use-cases.

Jupp, we will need need to fix unmanaged member function support for delegates in Mono to implement ComWrappers.

Reply:
Backport of https://github.com/mono/mono/pull/21365, https://github.com/dotnet/runtime/pull/64281/commits/38fa8ad7577d3fa05b1526173d9dbb293d7625e2